modules/cognito/secrets.mixin.tf

```diff
@@ -0,0 +1,75 @@
+# secrets.mixin.tf provides an abstract way for components to consume secret values from external sources
```

Comment:
We have decided to keep the mixins out of the component directories and instead put them in the root level [mixins](https://github.com/cloudposse/terraform-aws-components/tree/master/mixins) folder.

Reply:
@nitrocode that makes sense, but in this case... the component code needs to be written with usage of *abstract* secrets in mind (i.e. something that will provide the `local.secrets` values).  Maybe each component that utilizes secrets could then have a default `secrets.tf` that provided the `local.secrets` values and that file could be left out or overwritten when `vendor pull` happens. 

Thoughts on that? I'd be happy to write up a "Component Development Guide" in this repo which includes this pattern so we'd have a path forward that we can consolidate on?

Reply:
I think that can be built into the provider code. We really need a way to download a mixin and apply a patch or shell script through vendoring.

cc @aknysh @Nuru @osterman @mcalhoun

Reply:
@nitrocode + @osterman -- Friendly ping to see if this PR is of interest. Totally fine if not, but just realized it lost steam and wanted to bring it back up to either move it forward or close it.